english-helper/src/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,16 @@
+import { useEffect, useState } from "react";
+
+export function useLocalStorage(initialState: unknown, key: string) {
+  const [value, setValue] = useState(function () {
```

Comment:
also you can use try catch and add type

const [value, setValue] = useState<T>(() => {
  try {
    const storedValue = localStorage.getItem(key);
    return storedValue ? JSON.parse(storedValue) : initialState;
  } catch (error) {
    console.error("Error parsing localStorage value:", error);
    return initialState;
  }
});